src/Whoops/Run.php

```diff
@@ -31,17 +43,39 @@ final class Run implements RunInterface
      */
     private $handlerStack = [];
 
+    /**
+     * @var array
+     */
     private $silencedPatterns = [];
 
+    /**
+     * @var SystemFacade
+     */
     private $system;
 
+    /**
+     * In certain scenarios, like in shutdown handler, we can not throw exceptions.
+     *
+     * @var bool
+     */
+    private $canThrowExceptions = true;
+
+    /**
+     * Run constructor.
+     *
+     * @param SystemFacade|null $system
```

Comment:
in case where the php-native type already covers everything, the phpdoc itself is unnecessary/useless.

phpdoc-based-types should only be used when they add a value on top of the php-native type

Reply:
> in case where the php-native type already covers everything, the phpdoc itself is unnecessary/useless.

Yes, you're right. The phpdoc still useless especially if you use the scalar type-hinting in PHP7.

> phpdoc-based-types should only be used when they add a value on top of the php-native type

Whoops already uses phpdoc, I'm not adding something new.

Reply:
> Whoops already uses phpdoc, I'm not adding something new.

My comment was related to the line beeing changed here - not about other changes within this PR.

E.g. the phpdoc for this construct() here does not add value (in comparison to the php native types).

I dont say you shouldn‘t add phpdocs as a whole, but just for certain situations

Reply:
This makes sense.

Reply:
I've doc for  `$system` parameter.

Reply:
I've added a description for  `$system` parameter.